v2/tools/asoctl/internal/importing/arm_resource_importer_test.go

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package importing
+
+import (
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	"github.com/Azure/azure-service-operator/v2/api"
+	. "github.com/onsi/gomega"
+)
+
+func Test_ARMResourceImporter_GroupKindFromARMID(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		name          string
+		armId         string
+		expectedGroup string
+		expectedKind  string
+	}{
+		{
+			name:          "resource group",
+			armId:         "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/aso-rg",
+			expectedGroup: "resources.azure.com",
+			expectedKind:  "ResourceGroup",
+		},
+		{
+			name:          "storage account",
```

Comment:
You should test an extension resource like RoleAssignments. They're weird and their ARM IDs are weird. It's possible it'll throw off your logic so you wanna make sure you've got it working.

Reply:
I'm breaking things out into followup PRs; extension resources is one of them.